interview-questions.md

```diff
@@ -8,36 +8,86 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: This sounds like a One_to_Many type due to a Cohort having many students but each student only belongs to one cohort. In order to fix this, you would need to update the model of the student database buy adding the column for the foreign key and performing a migration.  
 
-Researched answer:
+Researched answer: Creating the migration file for the change is a good place to start here; $rails g migration add_foreign_key_to_students
+Then make sure in the method that uniqueness is true on the Cohort table and add foreign key to Student table.  Dont forget to $rails db:migrate.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: Well using params with the routes makes the overall app or web page more "dynamic".
 
-Researched answer:
+Researched answer:  Using params helps to not only make it overall more "Dynamic" but also lets you have more information, just create the method associated with your value.  Using params will help not only dynamically but reduce your over all scripting and perhaps the efficiency as well since it is data being sent back from the client to the servers, (sometimes in Idaho).
```

Comment:
This isn't wrong but it does not answer the question of which RESTful routes require params.